__integrations__/index.test.ts

```diff
@@ -0,0 +1,406 @@
+
+jest.setTimeout(WAIT_MS * 5);
+
+const baseUrl = "http://function:7071";
+
+beforeAll(async () => {});
+afterAll(async () => {});
+
+beforeEach(() => jest.clearAllMocks());
+
+// ----------------
+
+export const aLegacyInboxEnabledFiscalCode = "AAABBB01C02D345L" as FiscalCode;
+export const aLegacyInboxDisabledFiscalCode = "AAABBB01C02D345I" as FiscalCode;
+export const anAutoFiscalCode = "AAABBB01C02D345A" as FiscalCode;
+export const aManualFiscalCode = "AAABBB01C02D345M" as FiscalCode;
+
+export const anEnabledServiceId = "anEnabledServiceId" as NonEmptyString;
+export const aDisabledServiceId = "aDisabledServiceId" as NonEmptyString;
+const aNonExistingServiceId = "aNonExistingServiceId" as NonEmptyString;
+
+// ----------------
+
+export const aMessageBodyMarkdown = "test".repeat(80);
+export const aMessageContent: MessageContent = {
+  markdown: aMessageBodyMarkdown,
+  subject: "test".repeat(10)
+};
+
+// Must correspond to an existing serviceId within "services" colletion
+const aSubscriptionKey = "aSubscriptionKey";
+
+const customHeaders = {
+  "x-user-groups":
+    "ApiUserAdmin,ApiLimitedProfileRead,ApiFullProfileRead,ApiProfileWrite,ApiDevelopmentProfileWrite,ApiServiceRead,ApiServiceList,ApiServiceWrite,ApiPublicServiceRead,ApiPublicServiceList,ApiServiceByRecipientQuery,ApiMessageRead,ApiMessageWrite,ApiMessageWriteDefaultAddress,ApiMessageList,ApiSubscriptionsFeedRead,ApiInfoRead,ApiDebugRead,ApiMessageWriteEUCovidCert",
+  "x-subscription-id": anEnabledServiceId,
+  "x-user-email": "unused@example.com",
+  "x-user-id": "unused",
+  "x-user-note": "unused",
+  "x-functions-key": "unused",
+  "x-forwarded-for": "0.0.0.0",
+  "Ocp-Apim-Subscription-Key": aSubscriptionKey
+};
+
+const mockGetCustomHeaders = jest.fn(() => customHeaders);
+
+const mockNodeFetch = jest.fn(
+  async (input: RequestInfo, init?: RequestInit) => {
+    if (SHOW_LOGS) {
+      console.log("Sending request");
+      console.log(input);
+    }
+
+    const res = await ((nodeFetch as unknown) as typeof fetch)(input, {
+      ...init,
+      headers: {
+        ...(init?.headers ?? {}),
+        ...mockGetCustomHeaders()
+      }
+    });
+
+    if (SHOW_LOGS) {
+      console.log("Result: ");
+      console.log(res);
+    }
+
+    return res;
+  }
+);
+
+const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+const client = createClient<"SubscriptionKey">({
+  basePath: "/api/v1",
+  baseUrl,
+  fetchApi: mockNodeFetch,
+  withDefaults: op => params =>
+    op({
+      ...params,
+      SubscriptionKey: aSubscriptionKey
+    })
+});
+
+// Wait some time
+beforeAll(async () => {
+  let i = 0;
+  while (i < 100) {
+    console.log("Waiting the function to setup..");
+    try {
+      const response = await nodeFetch("http://function:7071/api/info");
+      break;
+    } catch (e) {
+      await delay(WAIT_MS);
+      i++;
+    }
+  }
+  if (i >= 10) {
+    console.log("Function unable to setup in time");
+    exit(1);
+  }
```

Comment:
I see this whole chunk of code is to handle function app startup time - which is something that should not be a responsibility of the test suite itself: tests should assume the environment is up&running.

I would not point that out if I didn't spot a flaw myself: if something is wrong, or even tests fail in unexpected manner, the while loop will continue to run even if Jest stopped, resulting in a flood of error message like 

> Cannot log after tests are done. Did you forget to wait for something async in your test?

I'd rather have a delay big enough (even 30s is acceptable to me): even in case it would not suffice, we would have a clear error reporting. Also, we can choose to use `sleep` bash command:
```sh
yarn start
sleep 30s
yarn test
```

It's not mandatory, though

Reply:
UPDATE: I found how to replicate my error (and why we are exposed to the flaw).

I was starting the test suit bad, thus my test agent wasn't able to connect to the function app, ever - thus iterating 100 times with 10s delay each.

Having Jest a default timeout of 25s on hooks execution, after the third failed attempt the worked ended, leaving 97 iterations to fail because I was logging after the test ended 🙃 .

Reply:
I know this is not good, but I was not able to speed up function container setup in code review pipeline. 
maybe I can set the maximum retry to 25 (and actually check the correct value -.-")

I also wanted to move to local .env file so we can decrease the default timeout

Reply:
I stick with my opinion: big enough `sleep` :)

Reply:
I refactor it a bit in [951e824](https://github.com/pagopa/io-functions-services/pull/167/commits/951e824e57f37152291038145b0d766aa764cd5c)